.circleci/config.yml

```diff
@@ -31,110 +32,182 @@ jobs:
       - checkout
 
       - run:
-          name: Install Dockerize
-          command: wget https://github.com/jwilder/dockerize/releases/download/$DOCKERIZE_VERSION/dockerize-linux-amd64-$DOCKERIZE_VERSION.tar.gz && sudo tar -C /usr/local/bin -xzvf dockerize-linux-amd64-$DOCKERIZE_VERSION.tar.gz && rm dockerize-linux-amd64-$DOCKERIZE_VERSION.tar.gz
+          name: "Apt: install system / build dependencies"
+          command: |
+            sudo apt-get update
+            sudo apt-get -y install \
+              curl \
+              gettext \
+              imagemagick \
+              libcurl4-gnutls-dev \
+              libexpat1-dev \
+              libssl-dev \
+              libz-dev \
+              postgresql-client
+
+      - type: cache-restore
+        name: "Dockerize: cache restore"
+        key: cache-{{ .Environment.CACHE_VERSION }}-dockerize-0.6.1
+
+      - run:
+          name: "Dockerize: install"
           environment:
             DOCKERIZE_VERSION: v0.6.1
+          command: |
+            dockerize_tar=dockerize-linux-amd64-${DOCKERIZE_VERSION}.tar.gz
+            if [[ ! -f ~/.cache/dockerize/${dockerize_tar} ]]; then
+              wget https://github.com/jwilder/dockerize/releases/download/${DOCKERIZE_VERSION}/${dockerize_tar}
+            fi
+            sudo tar -C /usr/local/bin -xzvf ${dockerize_tar}
+
+      - type: cache-save
+        name: "Dockerize: cache save"
+        key: cache-{{ .Environment.CACHE_VERSION }}-dockerize-0.6.1
+        paths:
+          - ~/.cache/dockerize
+
+
+      - type: cache-restore
+        name: "Git: cache restore"
```

Comment:
The updates to the CircleCI config add descriptive labels to caching steps, concentrate cache directories in `~/.cache`, and caches dependencies more effectively.